File: app/controllers/ForceHttps.java
package controllers;

import play.libs.F;
import play.mvc.Action;
import play.mvc.Controller;
import play.mvc.Http;
import play.mvc.Result;

public class ForceHttps extends Action<Controller> {

    @Override public F.Promise<Result> call(Http.Context ctx) throws Throwable {
        if (!ctx.request().secure()) {
            return F.Promise.promise(
                () -> redirect("https://" + ctx.request().host() + ctx.request().uri()));
        }
        return this.delegate.call(ctx);
    }
}

File: app/controllers/ForceHttps.kt
package controllers

import play.libs.F
import play.mvc.Action
import play.mvc.Controller
import play.mvc.Http
import play.mvc.Result

public class ForceHttps : Action<Controller>() {
    override fun call(ctx: Http.Context?): F.Promise<Result>? {
        if (ctx != null && !ctx.request().secure()) {
            return F.Promise.promise {
                redirect("https://${ctx.request().host()}${ctx.request().uri()}")
            }
        } else {
            return delegate.call(ctx)
        }

    }
}
